react/components/TabbedShelf/index.js

```diff
@@ -0,0 +1,156 @@
+import PropTypes from 'prop-types'
+import { injectIntl } from 'react-intl'
+import React, { Fragment, Component } from 'react'
+import { Button } from 'vtex.styleguide'
+import { withRuntimeContext } from 'vtex.render-runtime'
+import Tabs from './Tabs'
+import Shelf from '../../Shelf'
+
+const MAX_NUMBER_OF_MENUS = 6
+
+import tabbedShelf from './tabbedShelf.css'
+
+/**
+ * Tabbed Shelf Module
+ */
+class TabbedShelf extends Component {
+
+    /**
+     * The prop types for this component
+     * Used for type checking
+     */
+    static propTypes = {
+        isEnabled: PropTypes.bool,
+        headline: PropTypes.string,
+        bottomText: PropTypes.string,
+        tabs: PropTypes.PropTypes.arrayOf(PropTypes.shape({
+          id: PropTypes.number,
+          __editorItemTitle: PropTypes.string,
+        })),
+        shelf: PropTypes.object,
+    };
+
+    /**
+     * Default prop types for this component
+     */
+    static defaultProps = {
+        isEnabled: false,
+        headline: '',
+        bottomText: '',
+        tabs: [],
+        shelf: {},
+    };
+
+    /**
+     * Handle primary call to action button click
+     */
+    handleButtonClick(url) {
+        if(url != undefined && url != '') {
+            window.open(url.buttunUrl,"_self");
+        }
+    }
+
+    /**
+     * Return the component
+     *
+     * @returns {*}
+     */
+    render() {
+        const {
+            isEnabled,
+            headline,
+            bottomText,
+            buttunUrl,
+            buttonText,
+            tabs
+        } = this.props
+
+        const panes = (tabs.length > 0 ? tabs.map(tab => (
+            { menuItem: tab.__editorItemTitle, render: () => (
+              <Shelf
+                {...{ ...this.props.shelf }}
+                category={tab.id}
+              />
+            )
+            }
+        )) : '');
```

Comment:
i don't know if there's a reason for these `''` (here and below) but I think it'd be better to return `null` in these cases